job_bundles/Using_Files_In_Jobs.md

```diff
@@ -0,0 +1,1161 @@
+
+```
+Submitting to Queue: Q1
+...
+Hashing Attachments  [####################################]  100%
+Hashing Summary:
+    Processed 1 file totaling 39.0 B.
+    Skipped re-processing 0 files totaling 0.0 B.
+    Total processing time of 0.0327 seconds at 1.19 KB/s.
+
+Uploading Attachments  [####################################]  100%
+Upload Summary:
+    Processed 1 file totaling 39.0 B.
+    Skipped re-processing 0 files totaling 0.0 B.
+    Total processing time of 0.25639 seconds at 152.0 B/s.
+
+Waiting for Job to be created...
+Submitted job bundle:
+   job_attachments_devguide/
+Job creation completed successfully
+job-74148c13342e4514b63c7a7518657005
+```
+
+Two things happened here. First the `script.sh` file was hashed, and then it was uploaded to Amazon S3.
+
+Deadline Cloud's job attachments treats the S3 bucket as a [content-addressable storage](https://en.wikipedia.org/wiki/Content-addressable_storage).
+Your files are uploaded to objects in Amazon S3 whose name is derived from a [hash](https://en.wikipedia.org/wiki/Cryptographic_hash_function)
+of that file's contents. If two files have identical content, then they have the same hash value regardless of where the files are located or
+what they are named. This allows Deadline Cloud to avoid uploading a file if it has previously been uploaded and is still available. So,
+the first step is calculating a hash of the `script.sh` file and then checking S3 to determine whether or not the file had previously been uploaded.
+
+You can use the [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-welcome.html) to see the objects that were
+uploaded to Amazon S3:
+
+```bash
+# The name of queue `Q1`'s job attachments S3 bucket
+Q1_S3_BUCKET=$(
+  aws deadline get-queue --farm-id $FARM_ID --queue-id $QUEUE1_ID \
+    --query 'jobAttachmentSettings.s3BucketName' | tr -d '"'
+)
+
+aws s3 ls s3://$Q1_S3_BUCKET --recursive
+```
+
+Notice that two objects were uploaded to S3:
+
+1. `DeadlineCloud/Data/87cb19095dd5d78fcaf56384ef0e6241.xxh128` -- The contents of `script.sh`. The value `87cb19095dd5d78fcaf56384ef0e6241` in the
+   object key is the hash of the file's contents, and the extension `xxh128` indicates that the hash value was calculated as a 128 bit [xxhash](https://xxhash.com/).
+2. `DeadlineCloud/Manifests/<farm-id>/<queue-id>/Inputs/<guid>/a1d221c7fd97b08175b3872a37428e8c_input` -- A manifest object (described below) 
+   for the job submission. The values `<farm-id>`, `<queue-id>`, and `<guid>` are shorthanded here for brevity; they are your farm identifier,
+   queue identifier, and a random hexidecimal value, respectively. The value `a1d221c7fd97b08175b3872a37428e8c` in this example is a hash
+   value calculated from the string `/home/cloudshell-user/job_attachments_devguide`; the directory where `script.sh` is located.
+
+The manifest object contains all of the information for the input files ***from a specific root path*** 
```

Comment:
Dunno if we need to mention it, but there is also a local copy of the manifest created in the job_history directory for users to reference. See https://github.com/aws-deadline/deadline-cloud/blob/mainline/src/deadline/job_attachments/README.md#asset-manifests

Reply:
Interesting. I'm inclined to leave it out. Seems more in-place for a guide on the CLI than a more general one on job attachments.